contracts/zero-ex/contracts/src/features/native_orders/NativeOrdersSettlement.sol

```diff
@@ -514,6 +563,89 @@ abstract contract NativeOrdersSettlement is
+    /// @param order The Taker Signed RFQ order.
+    /// @param makerSignature The order signature from the maker.
+    /// @param takerSignature The order signature from the taker.
+    /// @return results Results of the fill.
+    function _fillTakerSignedRfqOrderPrivate(
+        LibNativeOrder.TakerSignedRfqOrder memory order,
+        LibSignature.Signature memory makerSignature,
+        LibSignature.Signature memory takerSignature
+    )
+        private
+        returns (FillNativeOrderResults memory results)
+    {
+        LibNativeOrder.TakerSignedOrderInfo memory takerSignedOrderInfo = getTakerSignedRfqOrderInfo(order);
+
+        // Must be fillable.
+        if (takerSignedOrderInfo.status != LibNativeOrder.OrderStatus.FILLABLE) {
+            LibNativeOrdersRichErrors.OrderNotFillableError(
+                takerSignedOrderInfo.orderHash,
+                uint8(takerSignedOrderInfo.status)
+            ).rrevert();
+        }
+
+        {
+            LibNativeOrdersStorage.Storage storage stor =
+                LibNativeOrdersStorage.getStorage();
+
+            // Must be fillable by the tx.origin.
+            if (order.txOrigin != tx.origin) {
+                LibNativeOrdersRichErrors.OrderNotFillableByOriginError(
+                    takerSignedOrderInfo.orderHash,
+                    tx.origin,
+                    order.txOrigin
+                ).rrevert();
+            }
+        }
+
+        // Signatures must be valid for the order.
+        {
+            address makerSigner = LibSignature.getSignerOfHash(takerSignedOrderInfo.orderHash, makerSignature);
+            if (makerSigner != order.maker) {
+                LibNativeOrdersRichErrors.OrderNotSignedByMakerError(
+                    takerSignedOrderInfo.orderHash,
+                    makerSigner,
+                    order.maker
+                ).rrevert();
+            }
+            address takerSigner = LibSignature.getSignerOfHash(takerSignedOrderInfo.orderHash, takerSignature);
+            if (takerSigner != order.taker) {
+                LibNativeOrdersRichErrors.OrderNotSignedByTakerError(
+                    takerSignedOrderInfo.orderHash,
+                    takerSigner,
+                    order.taker
+                ).rrevert();
+            }
+        }
+
+        // Settle between the maker and taker.
+        (results.takerTokenFilledAmount, results.makerTokenFilledAmount) = _settleTakerSignedOrder(
+            SettleTakerSignedOrderInfo({
+                orderHash: takerSignedOrderInfo.orderHash,
+                maker: order.maker,
+                taker: order.taker,
+                makerToken: IERC20TokenV06(order.makerToken),
+                takerToken: IERC20TokenV06(order.takerToken),
+                makerAmount: order.makerAmount,
+                takerAmount: order.takerAmount,
+                txOrigin: order.txOrigin,
+                txOriginNonce: order.txOriginNonce
+            })
```

Comment:
instead of this `SettleTakerSignedOrderInfo` struct, let's just pass in the orderHash and the `order`.